src/core/structure.js

```diff
@@ -202,11 +202,18 @@ function renameSectionHeaders() {
     return;
   }
   headers.forEach(header => {
-    const depth = Math.min(parents(header, "section").length + 1, 6);
-    const h = `h${depth}`;
+    const depth = parents(header, "section").length + 1;
+    const h = `h${Math.min(depth, 6)}`;
     if (header.localName !== h) {
       renameElement(header, h);
     }
+    if (depth > 6) {
+      if (header.classList) {
+        header.classList.add(`h${depth}`);
+      } else {
+        header.className = `h${depth}`;
```

Comment:
We don't need this as a special case as we support modern browsers (which support `Element.classList`), so the following is enough:
``` js
if (depth > 6) {
  header.classList.add(`h${depth}`);
}
```

Reply:
Fixed and updated.